### PR TITLE
feat: auto home handling

### PR DESCRIPTION
**Pull request by viceice**
- we now override `HOME=/home/$USER_NAME` if a random user is used which gets `HOME=/` by docker by default
- fixed / unified some env handling

### CODE REVIEW

File: src/usr/local/bin/install-buildpack [hunk 1]
```diff
@@ -28,7 +28,13 @@ fi
 
 # env helper, loads tool specific env
 cat >> $BASH_ENV <<- EOM
-export BUILDPACK=1 USER_NAME="${USER_NAME}" USER_ID="${USER_ID}"
+export BUILDPACK=1 USER_NAME="${USER_NAME}" USER_ID="${USER_ID}" USER_HOME="/home/${USER_NAME}"
+
+# openshift override unknown user home
+if [ "\${EUID}" != 0 ] && [ "\${HOME}" == "/" ]; then
```
Comment (rarkins) [inline src/usr/local/bin/install-buildpack]: Home is `/` if run as a random ID?

File: src/usr/local/buildpack/tools/dotnet.sh [hunk 2]
```diff
@@ -10,7 +10,7 @@ if [[ ! "${MAJOR}" || ! "${MINOR}" || ! "${PATCH}" ]]; then
   exit 1
 fi
 
-DOTNET_INSTALL_DIR=/usr/local/dotnet/${TOOL_VERSION}
+DOTNET_INSTALL_DIR=/usr/local/${TOOL_NAME}/${TOOL_VERSION}
```
Comment (rarkins) [inline src/usr/local/buildpack/tools/dotnet.sh:13]: Is there a scenario where this is not `dotnet` ?

File: src/usr/local/buildpack/tools/elixir.sh [hunk 3]
```diff
@@ -5,12 +5,17 @@ set -e
 require_root
 check_command erl
 
+if [[ -d "/usr/local/${TOOL_NAME}/${TOOL_VERSION}" ]]; then
+  echo "Skipping, already installed"
+  exit 0
+fi
```
Comment (rarkins) [inline src/usr/local/buildpack/tools/elixir.sh:11]: Is this specific to elixir? Seems like a generic rule. Easier copy/paste, less chance of human error?

File: src/usr/local/bin/install-buildpack [hunk 4]
```diff
@@ -28,7 +28,13 @@ fi
 
 # env helper, loads tool specific env
 cat >> $BASH_ENV <<- EOM
-export BUILDPACK=1 USER_NAME="${USER_NAME}" USER_ID="${USER_ID}"
+export BUILDPACK=1 USER_NAME="${USER_NAME}" USER_ID="${USER_ID}" USER_HOME="/home/${USER_NAME}"
+
+# openshift override unknown user home
+if [ "\${EUID}" != 0 ]; then
+  export HOME="\${USER_HOME}"
+fi
```
Comment (viceice) [inline src/usr/local/bin/install-buildpack:36]: this should makes it openshift compatible. So override home if we are not root

File: src/usr/local/bin/install-buildpack (hunk 1 above)
Comment (viceice) [reply src/usr/local/bin/install-buildpack]: yes, at least on docker. but removed that condition, so only root is checked

File: src/usr/local/buildpack/tools/dotnet.sh (hunk 2 above)
Comment (viceice) [reply src/usr/local/buildpack/tools/dotnet.sh:13]: not yet, just some simplification, so i can easily copy'n'past between tools 🙃

File: src/usr/local/buildpack/tools/elixir.sh (hunk 3 above)
Comment (viceice) [reply src/usr/local/buildpack/tools/elixir.sh:11]: not yet really happy with this, as for some tools i like to support multiple versions and latest install call should override env again, so executables in path always points to latest installed version.

see python or php for a more generic version. will extract some more helper functions later to make new tools even more easy to implement.

**Review by rarkins — APPROVED** [review]

**Pull request merged**